js/parts/DataLabels.js

```diff
@@ -60,23 +65,21 @@ var addEvent = H.addEvent,
     some = H.some,
     stableSort = H.stableSort;
 
-
 /**
  * General distribution algorithm for distributing labels of differing size
  * along a confined length in two dimensions. The algorithm takes an array of
  * objects containing a size, a target and a rank. It will place the labels as
  * close as possible to their targets, skipping the lowest ranked labels if
  * necessary.
  *
+ * @private
  * @function Highcharts.distribute
  *
- * @param  {Array<Highcharts.DataLabelBoxObject>} boxes
+ * @param {Array<Highcharts.DataLabelBoxObject>} boxes
```

Comment:
Too specific... It can be any box. Why do we need typing for this internal function in the first place?

Reply:
What does any box mean? This type only describes the minimum of required properties and corresponding types. This does not mean, that the provided objects may not contain other properties.

I type everything, to make sure, that the behavior of public functions using internal function is consistent with their doclet. Otherwise I would not know after several function calls what types are passing through the code and in the end are possible parameter types or return types.

Reply:
The `distribute` function, in addition to data labels, is used for legend items, split tooltips and flag series. So `DataLabelBoxObject` is too specific.

Reply:
Okay, I will reduce the doclets.